### PR TITLE
Update to main

### PR DESCRIPTION
**Pull request by phildini**

### CODE REVIEW

**Review by phildini — COMMENTED** [review]
Part 1

**Comment by MaggieFero** [comment]
I've started reviewing this, but I'm pausing for the night at 126/141 files viewed. Leaving this comment mostly so I can see if there's anything I need to look at again because of changes before I finish.

**Comment by MaggieFero** [comment]
> I've started reviewing this, but I'm pausing for the night at 126/141 files viewed. Leaving this comment mostly so I can see if there's anything I need to look at again because of changes before I finish.

No new commits since then! Picking up my review where I left off.

File: bookwyrm/views/admin/user_admin.py [hunk 1]
```diff
@@ -93,6 +95,10 @@ def post(self, request, user):
             form = forms.UserGroupForm(request.POST, instance=user)
             if form.is_valid():
                 form.save(request)
+
+        if report_id:
```
Comment (phildini) [inline bookwyrm/views/admin/user_admin.py:99]: It maybe sucks you can save the formgroup without recording the action.

**Review by MaggieFero — APPROVED** [review]
Overall, there are a couple things I don't love in here, but nothing we should block a merge over. There are a few security changes we've discussed OOB that I'd like us to test after merging but before deploying this, and if they work out for us we can also propose them upstream.

Overall, let's merge this!

**Pull request merged**